### PR TITLE
Add auth token details to SCAFFOLDING #17

### PR DESCRIPTION
**Pull request by shash-sgnl**

### CODE REVIEW

File: pkg/adapter/client.go
Comment (richardxgao) [inline pkg/adapter/client.go:48]: Hmm...we already have `Username` and `Password` as part of this `Request` struct. By adding `Token` we should remove the other authN method IMO.

Ultimately, does this `Token` need to be added? Isn't the scaffolding comment
```
// SCAFFOLDING #5 - pkg/adapter/client.go: Add/Remove/Update any fields to model the request for the SoR API.
```
sufficient? User would remove user/pass if they're using token auth.

Comment (shash-sgnl) [reply pkg/adapter/client.go:48]: If the SoR supports two methods - basic and OAuth, and the user decides to support both in the adapter, but use only one in the SoR (in SGNL) and maybe switch it up later in the Console, does that mean they have to make an adapter change if they want to switch auth types in the Sgnl console?

Comment (shash-sgnl) [reply pkg/adapter/client.go:48]: Another question for you: 
if the user chooses auth token in the sgnl console, ingestion will only send the token through to the adapter with username and password being empty strings? 
If later they move the SoR to use basic auth, ingestion will then send username/password and an empty string for token?

Comment (richardxgao) [reply pkg/adapter/client.go:48]: According to the adapter-framework, only [one of the auth mechanisms can be non-nil](https://github.com/SGNL-ai/adapter-framework/blob/f2cafb0d963b54c350350967906ce59776d720a1/adapter.go#L71-L82).

Ingestion respects that here: https://github.com/SGNL-ai/sgnl/blob/51bf54f9648a81d2706b2b26676c6aafe2af8594/pkg/ingestion_scheduler/workers/adapter_driver.go#L321-L331. Ingestion will only send one auth type at a time.

To answer your questions, yes I believe you can change the adapter so that it will use the non-empty auth mechanism. You will need to check `request.Auth.Basic == nil` or `request.Auth.HTTPAuthorization == nil` (only one of these can be non-nil), and then construct the SoR request accordingly. This change means if the user wants to switch auth mechanisms via the console at some point, it would not require an adapter code change.

So we'd have to change the template here: https://github.com/SGNL-ai/adapter-template/blob/a35b570aaf9e8ce39828757a9562339c09d1cc8d/pkg/adapter/adapter.go#L68-L75 as well.

Personally, I am on the side of keeping the template lean and simple, but not a strong feeling.

cc @nholbrook @r-rakshith for further thoughts.

Comment (shash-sgnl) [reply pkg/adapter/client.go:48]: I think this will keep it lean and simple + users do not have to worry about implementing either auth mechanism anymore.

Comment (shash-sgnl) [reply pkg/adapter/client.go:48]: @richardxgao Take a look now. 
Added if-then to check for token and/or basic, and use whichever is being sent by ingestion. 

@r-rakshith @nholbrook This reduces the amount of code a user will need to write by a bit.

Comment (r-rakshith) [reply pkg/adapter/client.go:48]: +1 on simplifying the work for the user. I think we should add comments and also group together the fields based on auth mechanism i.e. username/password under the basic auth comment and token under a different comment group. WDYT?

File: pkg/adapter/datasource.go
Comment (r-rakshith) [inline pkg/adapter/datasource.go:114]: Let's not create intermediate variables that'll only be used once.

```suggestion
		req.Header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(auth)))
```

File: pkg/adapter/datasource.go
Comment (r-rakshith) [inline pkg/adapter/datasource.go:113]:
```suggestion
```

**Review by r-rakshith — CHANGES_REQUESTED** [review]
Looks good overall. Couple of minor changes requested.

File: pkg/adapter/datasource.go
Comment (shash-sgnl) [reply pkg/adapter/datasource.go]: Done

File: pkg/adapter/client.go
Comment (shash-sgnl) [reply pkg/adapter/client.go:48]: I've clubbed the variables together under the two comments.

File: pkg/adapter/datasource.go
Comment (r-rakshith) [inline pkg/adapter/datasource.go:113]: nit: I understand that we might not want to setup CI with linters on a public repo. Let's make sure to run them locally if possible before merging. Not a blocker right now. We (@r-rakshith / @richardxgao / @nholbrook) will follow up in a different PR.

```suggestion
		req.Header.Add("Authorization", "Basic " + base64.StdEncoding.EncodeToString([]byte(auth)))
```

File: pkg/adapter/adapter.go
Comment (r-rakshith) [inline pkg/adapter/adapter.go:76]: nit: styling
```suggestion
		Token:            request.Auth.HTTPAuthorization,
		
```

**Review by r-rakshith — APPROVED** [review]

File: pkg/adapter/client.go
Comment (richardxgao) [reply pkg/adapter/client.go:48]: @shash-sgnl LGTM.

**Review by richardxgao — APPROVED** [review]
LGTM once styling comments are addressed.

**Pull request merged**
